cumulus_library/template_sql/statistics/psm_templates.py

```diff
@@ -0,0 +1,87 @@
+""" Collection of jinja template getters for common SQL queries """
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Optional
+
+from jinja2 import Template
+from pandas import DataFrame
+
+from cumulus_library.errors import CumulusLibraryError
+
+
+def get_distinct_ids(
+    columns: list[str], source_table: str, join_id: str = None, filter_table: str = None
+) -> str:
+    """Gets distinct ids from a table, optionally filtering by ids in another table
+
+    This is expected to be used in two ways:
+    - To retrieve ids from a study cohort table
+    - To retreive ids from a core FHIR table, excluding ids in a cohort study table
+
+    It is expected that, if supplying optional parameters, all of them must be set
+
+    :param columns: a list of ids to request
+    :param source_table: the table to retrieve ids from
+    :param join_id: the id column to use for joining. Expected to exist in both source and filter tables
+    :param filter_table: a table containing ids you want to exclude
+    """
+    if (join_id is None and filter_table is not None) or (
+        join_id is not None and filter_table is None
+    ):
```

Comment:
nit: Do you want to allow empty strings here? If not, this may be more readable with straight boolean logic:
```
if (join_id and not filter_table) or (not join_id and filter_table):
```
or
```
if bool(join_id) != bool(filter_table):
```

(same pattern below with count table)

Reply:
ok, can do